frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/__tests__/k8s-ref.spec.ts

```diff
@@ -0,0 +1,72 @@
+import { DeploymentModel } from '../__mocks__/k8s-data';
+import {
+  getAPIVersionForModel,
+  getGroupVersionKindForReference,
+  getGroupVersionKindForResource,
+  getReference,
+  getReferenceForModel,
+} from '../k8s-ref';
+
+describe('k8s-Resource', () => {
+  it('should return reference for provided group, version, and kind', () => {
+    const referenceData = getReference({ group: 'apps', version: 'v1', kind: 'Deployment' });
+    expect(referenceData).toEqual('apps~v1~Deployment');
+  });
+
+  it('should return reference for provided version and kind', () => {
+    const referenceData = getReference({ group: null, version: 'v1', kind: 'Pod' });
+    expect(referenceData).toEqual('core~v1~Pod');
+  });
+
+  it('should return reference for provided model', () => {
+    const referenceData = getReferenceForModel(DeploymentModel);
+    expect(referenceData).toEqual('apps~v1~Deployment');
+  });
+
+  it('should return apiVersion for provided model', () => {
```

Comment:
Did we want a test for when `apiGroup` is falsy?

Reply:
make sense, added